src/ast/passes/codegen_llvm.cpp

```diff
@@ -150,16 +150,91 @@ void CodegenLLVM::kstack_ustack(const std::string &ident,
                                 StackType stack_type,
                                 const location &loc)
 {
-  Value *stackid = b_.CreateGetStackId(
-      ctx_, ident == "ustack", stack_type, loc);
+  if (!murmur_hash_2_func_)
+    murmur_hash_2_func_ = createMurmurHash2Func();
+
+  const bool is_ustack = ident == "ustack";
+  const auto uint64_size = sizeof(uint64_t);
+
+  AllocaInst *stackid = b_.CreateAllocaBPF(CreateUInt64(), "stackid");
```

Comment:
We'll need to add `b_.CreateLifetimeEnd(stackid);` after the last usage to avoid leaking the BPF memory